src/state/slices/stakingDataSlice/stakingDataSlice.ts

```diff
@@ -0,0 +1,89 @@
+import { createSlice } from '@reduxjs/toolkit'
+import { createApi, fetchBaseQuery } from '@reduxjs/toolkit/query/react'
+import { CAIP19, caip19 } from '@shapeshiftoss/caip'
+import { chainAdapters, ChainTypes } from '@shapeshiftoss/types'
+import { getChainAdapters } from 'context/PluginProvider/PluginProvider'
+
+export type PubKey = string
+type AllStakingDataArgs = { pubKey: string; assetId: CAIP19 }
+
+export type Staking = {
+  delegations: chainAdapters.cosmos.Delegation[]
+  redelegations: chainAdapters.cosmos.Redelegation[]
+  undelegations: chainAdapters.cosmos.Undelegation[]
+  rewards: chainAdapters.cosmos.Reward[]
+}
+export type StakingDataById = {
+  [k: PubKey]: Staking
+}
+
+export type StakingData = {
+  byPubKey: StakingDataById
+}
+
+export type StakingPayload = {
+  payload: {
+    pubKey: PubKey
+    stakingData: Staking
+  }
+}
+const initialState: StakingData = {
+  byPubKey: {}
+}
+
+const updateOrInsert = (stakingDataState: StakingData, pubKey: PubKey, currentStakingData: any) => {
+  stakingDataState.byPubKey[pubKey] = currentStakingData
```

Comment:
Elsewhere we tend to use accountSpecifier for the key, as opposed to pubkey. This is essentially `${caip2}:${pubkey}` and gives us a bit more information in the key to identify the chain.

Functionally, this works the same when it's only cosmos, but I can see down the line when ETH staking is in there, having that caip2 would be valuable vs parsing the pubkey to determine the chain/network.